src/fieri/app/models/license_worker.rb

```diff
@@ -0,0 +1,43 @@
+require 'sidekiq'
+require 'net/http'
+
+class LicenseWorker
+  include ::Sidekiq::Worker
+
+  # Acceptable licenses were determined and documented in
+  # https://github.com/chef-cookbooks/cookbook-quality-metrics/blob/master/quality-metrics/qm-003-license.md
+  ACCEPTABLE_LICENSES = ['Apache 2.0', 'MIT', 'GNU Public License 2.0', 'GNU Public License 3.0'].freeze
+
+  def perform(version_json, cookbook_name)
+    Net::HTTP.post_form(
+      URI.parse("#{ENV['FIERI_SUPERMARKET_ENDPOINT']}/api/v1/cookbook-versions/license_evaluation"),
+      fieri_key: ENV['FIERI_KEY'],
+      cookbook_name: cookbook_name,
+      cookbook_version: JSON.parse(version_json)['version'],
+      license_failure: license_failure?(version_json),
+      license_feedback: license_feedback(cookbook_name, license_failure?(version_json))
+    )
+  end
+
+  private
+
+  def license_failure?(version_json)
+    license = JSON.parse(version_json)['license']
+
+    ACCEPTABLE_LICENSES.include?(license) ? false : true
+  end
+
+  def license_feedback(cookbook_name, failure)
+    failure == true ? "#{cookbook_name} needs a valid open source license\n#{acceptable_licenses_string}" : ''
+  end
+
+  def acceptable_licenses_string
```

Comment:
I think this could be made a little bit simpler.

```ruby
def acceptable_licenses_string
  "Acceptable licenses include #{ACCEPTABLE_LICENSES.join(', ')}."
end
```

```
[1] pry(main)> ACCEPTABLE_LICENSES = ['Apache 2.0', 'MIT', 'GNU Public License 2.0', 'GNU Public License 3.0']
=> ["Apache 2.0", "MIT", "GNU Public License 2.0", "GNU Public License 3.0"]
[2] pry(main)> "Acceptable licenses include #{ACCEPTABLE_LICENSES.join(', ')}."
=> "Acceptable licenses include Apache 2.0, MIT, GNU Public License 2.0, GNU Public License 3.0."
```

If the building of the string is reduced to that, it could even get moved back into `#license_feedback()`. And maybe a positive version of the feedback string could be provided:

```ruby
def license_feedback(cookbook_name, failure)
  if failure
    "#{cookbook_name} does not have a valid open source license.\nAcceptable licenses include #{ACCEPTABLE_LICENSES.join(', ')}."
  else
    "#{cookbook_name} has a recognized open source license."
  end
end
```